public-ui/src/app/highlight-search/highlight-search.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-highlight-search',
+  templateUrl: './highlight-search.component.html',
+  styleUrls: ['./highlight-search.component.css']
+})
+export class HighlightSearchComponent implements OnInit, OnChanges {
+  @Input() text: string;
```

Comment:
A lot of these inputs aren't currently used - are they all needed?

Reply:
They are used in different contexts. In case the search involves the display of standard / source concepts and such.